core/aws-core/src/main/java/software/amazon/awssdk/awscore/internal/protocol/json/JsonErrorCodeParser.java

```diff
@@ -60,7 +74,30 @@ public String parseErrorCode(SdkHttpFullResponse response, JsonContent jsonConte
      * present in the header.
      */
     private String parseErrorCodeFromHeader(SdkHttpFullResponse response) {
-        String headerValue = response.firstMatchingHeader(X_AMZN_ERROR_TYPE).orElse(null);
+        Map<String, List<String>> filteredHeaders = response.headers().entrySet().stream()
+                                                            .filter(e -> errorCodeHeaders.contains(e.getKey()))
+                                                            .collect(Collectors.toMap(Map.Entry::getKey, Map.Entry::getValue));
+
+        if (filteredHeaders.isEmpty()) {
+            return null;
+        }
+
+        if (filteredHeaders.size() > 1) {
```

Comment:
What is the customer experience in this situation? We generally try to be as resilient as possible when unmarshlaling the exceptions so we can preserve as much info as possible about the error.

Reply:
Are you saying we should combine all the error codes? I think its confusing if service multiple headers representing the error code with different values. If they do, maybe we can select the first option and return it? what do you think?

Reply:
Whatever we do we should try to fail as gracefully a possible so that we still generate a service exception rather than a runtime one. I'm thinking selecting the first one we find and log a warning.